executor/common_bsd.h

```diff
@@ -255,17 +255,16 @@ static long syz_extract_tcp_res(long a0, long a1, long a2)
 	return 0;
 }
 #endif
+#endif // GOOS_openbsd
 
 #if SYZ_EXECUTOR || SYZ_SANDBOX_NONE
 static void loop();
 static int do_sandbox_none(void)
 {
-#if SYZ_EXECUTOR || SYZ_TUN_ENABLE
+#if SYZ_TUN_ENABLE
```

Comment:
This is not correct. executor build does not define SYZ_TUN_ENABLE or any similar macros, it only defines SYZ_EXECUTOR and this needs to enable everything. With this change tun will never be enabled during fuzzing.
For executor we need to disable features dynamically. Or, for this case I guess based on OS macros, e.g.: `#if GOOS_openbsd && (SYZ_EXECUTOR || SYZ_TUN_ENABLE)`.
Maybe it's time to split common_bsd.h into 3 files per-OS to make such breakages less frequent, but they all can include common header still.

@blackgnezdo @mptre  for openbsd
@msvoelker for freebsd

As expected as soon as we stopped testing freebsd build on linux, it broke immediately.

Reply:
Thanks for the suggestion. Fixed in https://github.com/google/syzkaller/pull/835/commits/1d9616f8160ef52318d7165e654300a9a2bda46e.
My plan is to add support for the tun interfaces to FreeBSD soon, just wanted to get it working right without it. With the changes up to now I was already able to find a bug in FreeBSD...